services/chainstorage/src/main/java/tech/pegasys/teku/services/chainstorage/StorageService.java

```diff
@@ -117,24 +118,28 @@ protected SafeFuture<?> doStart() {
                             pruningActiveLabelledGauge));
               }
               if (config.getDataStorageMode().storesFinalizedStates()
-                  && config.getRetainedSlots() > -1) {
-                LOG.info(
-                    "State pruner will run every: {} minute(s), retaining states for the last {} finalized slots. Limited to {} state prune per execution. ",
-                    config.getStatePruningInterval().toMinutes(),
-                    config.getRetainedSlots(),
-                    config.getStatePruningLimit());
-                statePruner =
-                    Optional.of(
-                        new StatePruner(
-                            config.getSpec(),
-                            database,
-                            storagePrunerAsyncRunner,
-                            config.getStatePruningInterval(),
-                            config.getRetainedSlots(),
-                            config.getStatePruningLimit(),
-                            "state",
-                            pruningTimingsLabelledGauge,
-                            pruningActiveLabelledGauge));
+                  && config.getRetainedSlots() > 0) {
+                if (config.getDataStorageCreateDbVersion() == DatabaseVersion.LEVELDB_TREE) {
+                  LOG.warn("State pruning is not supported with leveldb_tree database.");
```

Comment:
it's basically invalid configuration exception. i think we just barf here, and make the user fix the configuration... I think potentially just throwing an InvalidConfigurationException, which iirc would terminate the client ultimately...
My reasoning is basically that this will never work...
Come to think of it the historical batch builder and this pruner are explicitly non compatible too, not sure if we've handled that...

Reply:
It makes sense to bail at this stage, agreed. Changed it to do so.